scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,484 @@
+
+This plot will also show, for each cell, the posterior probability that the cell is derived from the compromised distribution; a higher score indicates that a cell is more likely to be compromised.
+
+It is also critical to note that this model can _and does_ fail at times. 
+Plotting the results as we have done here is not a step to skip. 
+**Always look at your data!**
+
+```{r miQC plotModel}
+miQC::plotModel(filtered_sce, miqc_model) + theme_bw()
+```
+
+We can now filter our data based on the probability compromised as calculated from the model.
+But before we do that, we might want to quickly plot to see what would be filtered out with a given cutoff.
+The default is to exclude cells that have a posterior probability of 0.75 or greater of being compromised.
+
+
+```{r miQC plotFiltering}
+miQC::plotFiltering(filtered_sce, miqc_model, posterior_cutoff = 0.75)
+```
+
+In this case seems to correspond to a mitochondrial percentage of about 12.5%, but note that this will not always be constant: the cutoff point can vary for different numbers of unique genes.
+
+At this point, we can perform the actual filtering using the `filterCells()` function, giving us a further filtered SCE object.
+
+```{r miQC filtercells}
+qcfiltered_sce <- miQC::filterCells(filtered_sce, model = miqc_model)
+```
+
+### One more filter: unique genes
+
+While the miQC filtering is pretty good, you may have noticed that it still left some cells that had very low numbers of unique genes. 
+While these cells may not be compromised, the information from them is also not likely to be useful, so we will filter those as well. 
+For now, we will use a cutoff of 200 unique genes as a minimum. 
+
+```{r unique cutoff}
+qcfiltered_sce <- qcfiltered_sce[, which(qcfiltered_sce$detected >= 200)]
+```
+
+
+
+## Normalization and Dimension Reduction
+
+Now that we have done our filtering, we can set analyzing the expression counts for the remaining cells. 
+
+The next step at this point is to convert the raw counts into a measure that accounts for differences in sequencing depth between cells, and to convert the distribution of expression values from the skewed distribution we expect to see in raw counts to one that is more normally distributed.
+
+We will do this using functions from the `scran` and `scuttle` packages. 
+The procedure we will use here is derived from [Orchestrating Single-Cell Analysis with Bioconductor](http://bioconductor.org/books/3.16/OSCA.basic/normalization.html#normalization-by-deconvolution).
+The idea is to first account for the fact that different cell types may have different expression patterns that could affect the normalization factors we would calculated if we were to use the raw sequencing depth. To account for that, we first do a rough clustering of cells, then use that clustering to calculate the scaling factor (sum factor) for each cell within the cluster.
```

Comment:
OSPL. Also this was all a bit tricky for me to read so I've suggested the following, take what you think works! Also note I use only "scaling factor" and not "normalization factor" to avoid hopping back and forth, since the code comments use "scaling" - 

```suggestion
The approach helps account for expression differences among different cell types, which can affect the scaling factors calculated when using the raw sequencing depth. 
To ameliorate problems resulting from differences in sequencing depth, we perform a rough clustering of cells (`scran::quickCluster()`), and then use that clustering to calculate the scaling factor (`scran::computeSumFactors()`) for each cell within its cluster.
```